libr/core/canal.c

```diff
@@ -819,8 +823,20 @@ static int __core_anal_fcn(RCore *core, ut64 at, ut64 from, int reftype, int dep
 		} else if (fcnlen == R_ANAL_RET_END) { /* Function analysis complete */
 			f = r_core_flag_get_by_spaces (core->flags, fcn->addr);
 			if (f && f->name && strncmp (f->name, "sect", 4)) { /* Check if it's already flagged */
-				R_FREE (fcn->name);
-				fcn->name = strdup (f->name);
+				const char *new_name = f->name;
```

Comment:
what about using the rbin api instead of the flags one?

Reply:
Do you mean `r_bin_get_symbols()`?

Reply:
yes. its O(n) but we should have O(1) apis in rbin at some point imho